From afaf2a9479e3c7dc668397aeed10bc5c327489ab Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Thu, 07 Mar 2024 12:55:38 +0200
Subject: [PATCH] [SW-156330] gaudi2: tpm: remove second tpm init

In gaudi2, tpm_init is called twice. During the PCR
extension for ppboot, tpm_init is called initially,
and subsequently, a second call happens during
the PCR extension of preboot. The primary purpose of
the tpm_init function is to discover the TPM device,
making the second call redundant. Therefore, removing
the second tpm_init call.

Change-Id: If9557ddaf72145b46ecc535fd2eb589183cbc907
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 837f726..1a0dc9a 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -418,18 +418,14 @@
 	 * Hash is read and not calculated because of ppboot image size
 	 * limitations
 	 */
-	ret = tpm_init(true);
-	if (!ret) {
-		if (magic != PREBOOT_SHA_MAGIC)
-			preboot_puts("no preboot sha magic found\n");
-		else
-			memcpy(hash, footer_offset + PREBOOT_HASH_MAGIC_SIZE, SHA384_SUM_LEN);
 
-		if (tpm_pcr_extend(TPM2_PREBOOT_PCR_IDX, hash))
-			preboot_puts("PPBOOT: pcr extend failed\n");
-	} else if (ret == -EOPNOTSUPP) {
-		preboot_puts("PPBOOT: TPM is not supported\n");
-	}
+	if (magic != PREBOOT_SHA_MAGIC)
+		preboot_puts("no preboot sha magic found\n");
+	else
+		memcpy(hash, footer_offset + PREBOOT_HASH_MAGIC_SIZE, SHA384_SUM_LEN);
+
+	if (tpm_pcr_extend(TPM2_PREBOOT_PCR_IDX, hash))
+		preboot_puts("PPBOOT: pcr extend failed\n");
 
 	return ARCBOOT_BRINGUP_OK;
 }
